### PR TITLE
build spades from source

### PR DESCRIPTION
**Pull request by notestaff**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [X] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/spades/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
+  url: http://cab.spbu.ru/files/release{{ version }}/SPAdes-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - 0001-change-verbosity.patch
+    - 0002-disable-parallel-libstdc.patch
 
 build:
-  number: 0
+  number: 1
+  skip: True  # [linux]
```
Comment (dpryan79) [inline recipes/spades/meta.yaml]: I presume this skip is just for testing purposes :)

**Comment by notestaff** [comment]
@bioconda/core I'm getting a lint failure should_not_be_noarch_source even though the recipe does not say it is noarch: https://circleci.com/gh/bioconda/bioconda-recipes/70110
Would appreciate any help on understanding this.  Thanks!

**Comment by dpryan79** [comment]
@notestaff Yeah, that lint has a high false-positive rate, just add a skip-lint for it. I'll update a related issue on bioconda-utils to mention this (https://github.com/bioconda/bioconda-utils/issues/598 ).

File: recipes/spades/meta.yaml [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
 
 build:
-  number: 0
+  number: 1
 
 requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - libcxx # [osx]
```
Comment (dpryan79) [inline recipes/spades/meta.yaml]: Does this really need libcxx?

Comment (dpryan79) [reply recipes/spades/meta.yaml]: Actually, I think `{{ compiler('cxx') }}` pulls that in. I'll play around with this a bit and revert & merge if I break something :)

**Comment by notestaff** [comment]
See also https://github.com/ablab/spades/issues/194#issuecomment-522768578 ; I think the patch I have here is functionally equivalent.

**Comment by dpryan79** [comment]
Nicely done @notestaff, if I didn't muck anything up in my changes then please merge at your leisure :)

**Comment by notestaff** [comment]
Looks good, thanks @dpryan79 !

**Comment by notestaff** [comment]
@bioconda/core Am I correct that there's no way to build a bioconda package for osx using gcc?  (Context: https://github.com/ablab/spades/issues/194#issuecomment-523175204 )

**Comment by dpryan79** [comment]
Correct

File: recipes/spades/meta.yaml [hunk 3]
```diff
@@ -1,22 +1,41 @@
...
+    - llvm-openmp  # [osx]
+    - clangdev  # [osx]
+    - openmp  # [linux]
+    - cmake
+  host:
+    - llvm-openmp  # [osx]
+    - zlib
+    - bzip2
+    - openmp  # [linux]
+    - jemalloc
```
Comment (asl) [inline recipes/spades/meta.yaml]: You don't need it. We're using integrated one.

File: recipes/spades/meta.yaml [hunk 4]
```diff
@@ -1,22 +1,41 @@
...
+    - readline
+    - ncurses
   run:
+    - llvm-openmp  # [osx]
+    - openmp  # [linux]
+    - jemalloc
```
Comment (asl) [inline recipes/spades/meta.yaml]: You don't need it. We're using integrated one.

File: recipes/spades/meta.yaml [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
+    - clangdev  # [osx]
+    - openmp  # [linux]
+    - cmake
+  host:
+    - llvm-openmp  # [osx]
+    - zlib
+    - bzip2
+    - openmp  # [linux]
+    - jemalloc
+    - readline
```
Comment (asl) [inline recipes/spades/meta.yaml]: Neither readline nor ncurses are necessary for SPAdes. Only for some dev tools.

Comment (notestaff) [reply recipes/spades/meta.yaml]: Ok -- added because cmake was checking for them, will remove.

**Pull request merged**